### PR TITLE
 [WARNING] CS0628: `new protected member declared in sealed type`

### PR DESCRIPTION
**Pull request by eaba**
## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: src/core/Akka.Remote/Endpoint.cs [hunk 1]
```diff
@@ -361,7 +361,7 @@ public EndpointAssociationException(string message)
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext"/> that contains contextual information about the source or destination.</param>
-        protected EndpointAssociationException(SerializationInfo info, StreamingContext context)
+        EndpointAssociationException(SerializationInfo info, StreamingContext context)
```
Comment (eaba) [inline src/core/Akka.Remote/Endpoint.cs:364]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka.Remote/Endpoint.cs [hunk 2]
```diff
@@ -386,7 +386,7 @@ public OversizedPayloadException(string message)
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo"/> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext"/> that contains contextual information about the source or destination.</param>
-        protected OversizedPayloadException(SerializationInfo info, StreamingContext context)
+        OversizedPayloadException(SerializationInfo info, StreamingContext context)
```
Comment (eaba) [inline src/core/Akka.Remote/Endpoint.cs:389]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka.Streams/ActorMaterializer.cs [hunk 3]
```diff
@@ -347,7 +347,7 @@ public AbruptStageTerminationException(GraphStageLogic logic)
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-        protected AbruptStageTerminationException(SerializationInfo info, StreamingContext context) : base(info, context) { }
+        AbruptStageTerminationException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (eaba) [inline src/core/Akka.Streams/ActorMaterializer.cs]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka.Streams/Dsl/Graph.cs [hunk 4]
```diff
@@ -1219,7 +1219,7 @@ public PartitionOutOfBoundsException(string message) : base(message)
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-        protected PartitionOutOfBoundsException(SerializationInfo info, StreamingContext context) : base(info, context) { }
+        PartitionOutOfBoundsException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (eaba) [inline src/core/Akka.Streams/Dsl/Graph.cs]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka.Streams/Dsl/Tcp.cs [hunk 5]
```diff
@@ -309,7 +309,7 @@ public TcpIdleTimeoutException(string message, TimeSpan duration) : base(message
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-        protected TcpIdleTimeoutException(SerializationInfo info, StreamingContext context) : base(info, context) { }
+        TcpIdleTimeoutException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```
Comment (eaba) [inline src/core/Akka.Streams/Dsl/Tcp.cs]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka/Actor/Scheduler/SchedulerException.cs [hunk 6]
```diff
@@ -26,7 +26,7 @@ public sealed class SchedulerException : AkkaException
         /// </summary>
         /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
         /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-        protected SchedulerException(SerializationInfo info, StreamingContext context)
+        SchedulerException(SerializationInfo info, StreamingContext context)
```
Comment (eaba) [inline src/core/Akka/Actor/Scheduler/SchedulerException.cs]: I have removed that `protected` @Aaronontheweb

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveStreams.Net.verified.txt [hunk 7]
```diff
@@ -1804,7 +1803,6 @@ namespace Akka.Streams.Dsl
     public sealed class PartitionOutOfBoundsException : System.Exception
     {
         public PartitionOutOfBoundsException(string message) { }
-        protected PartitionOutOfBoundsException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
```
Comment (eaba) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveStreams.Net.verified.txt:1807]: removed @Aaronontheweb

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveStreams.Net.verified.txt [hunk 8]
```diff
@@ -12,7 +12,6 @@ namespace Akka.Streams
     public sealed class AbruptStageTerminationException : System.Exception
     {
         public AbruptStageTerminationException(Akka.Streams.Stage.GraphStageLogic logic) { }
-        protected AbruptStageTerminationException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
```
Comment (eaba) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveStreams.Net.verified.txt:15]: removed @Aaronontheweb

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt [hunk 9]
```diff
@@ -1614,7 +1614,6 @@ namespace Akka.Actor
     public sealed class SchedulerException : Akka.Actor.AkkaException
     {
         public SchedulerException(string message) { }
-        protected SchedulerException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
```
Comment (eaba) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt:1617]: removed @Aaronontheweb

**Review by eaba — COMMENTED** [review]
@Aaronontheweb changes

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt (hunk 9 above)
Comment (Aaronontheweb) [reply src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.DotNet.verified.txt:1617]: I think that needs to be made public, probably

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.Net.verified.txt [hunk 10]
```diff
@@ -1612,7 +1612,6 @@ namespace Akka.Actor
     public sealed class SchedulerException : Akka.Actor.AkkaException
     {
         public SchedulerException(string message) { }
-        protected SchedulerException(System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context) { }
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.Net.verified.txt:1615]: Make all of these public

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Need the serialization constructors back, so make them `public` instead

**Comment by eaba** [comment]
Changes to `public` @Aaronontheweb

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**